packages/pcd/eddsa-pcd/src/EDDSAPCDPackage.ts

```diff
@@ -11,32 +22,39 @@ import {
   EdDSAPCDTypeName,
   EdDSAPublicKey
 } from "./EdDSAPCD";
-
-let initializedPromise: Promise<void> | undefined;
-let eddsa: Eddsa;
+import { encodePublicKey, publicKeyToPoint } from "./util/util";
 
 /**
- * A promise designed to make sure that the EdDSA algorithm
- * of the `circomlibjs` package has been properly initialized.
- * It only initializes them once.
+ * Generates a Poseidon hash from an array of bigint values.
+ * In order to avoid including unnecessary Poseidon constants, we only support
+ * arrays of certain lengths, based on usage by the dependents of EdDSAPCD.
```

Comment:
That's reasonable.  Given this can only sign an array of BigInts, real "message' use cases will probably want to do their own hashing of a string or something similar.  Maybe document that assumption, and the specific lengths currently supported?  I like the inline comments specifying which dependent it's tied to.